package/sbp_settings_daemon/sbp_settings_daemon/src/settings.c

```diff
@@ -288,66 +280,73 @@ static void settings_save_callback(u16 sender_id, u8 len, u8 msg[], void *contex
   fclose(f);
 }
 
-static void settings_write_callback(u16 sender_id, u8 len, u8 msg[], void *context)
+static void settings_write_reject(sbp_tx_ctx_t *tx_ctx,
```

Comment:
per the below comment this might be simplified into two reject subroutines, where in the case of a failed parse, `SETTINGS_WR_PARSE_FAILED` could be returned as well as simply `memcpy`-ing the setting buffer as is back to the requestor. If we can't decode the entirety of the buffer, then it doesn't make much sense to re-encode only what we could partially decode. thoughts?

Reply:
OK, I see what you mean. I'll work something out.